src/unix/unix_internal.h

```diff
@@ -390,6 +390,7 @@ typedef struct fdesc {
     int type;
     int flags;                  /* F_GETFD/F_SETFD flags */
     notify_set ns;
+    void *private;
```

Comment:
Wouldn't it be better to put this field in struct file instead? fdesc is used as a base struct for all types of file descriptors, so I think private data specific for a given descriptor type should be embedded in the struct that extends this base struct for that descriptor type.

Reply:
You're right, in fact I decided to rework the special file allocation so that no new fields needed to be added to file or fdesc structs, so special files can now have their own struct containing a file struct.